src/theory/theory_engine.h

```diff
@@ -363,6 +363,7 @@ class TheoryEngine {
   /** Constructs a theory engine */
   TheoryEngine(context::Context* context,
                context::UserContext* userContext,
+               ResourceManager* rm,
```

Comment:
I'm not sure about doing this, since it makes the conceptual dependencies circular (TheoryEngine <-> SmtEngine).  Let's think about this  more after this PR.

Reply:
I am not generally a fan of circular dependencies but I think that `SmtEngine` (or some similar object) will be one of the few exceptions in the future. If we want to get rid of all the singletons then we have to have some object that grants you access to all the "global" information.

Reply:
I agree with @4tXJ7f .

Reply:
To be addressed in a later PR.